PennMobile/LaundryPrototype/ViewModels/LaundryViewModel.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  LaundryViewModel.swift
+//  PennMobile
+//
+//  Created by Nathan Aronson on 10/19/25.
+//  Copyright © 2025 PennLabs. All rights reserved.
+//
+
+import Foundation
+import PennMobileShared
+import SwiftUI
+
+@MainActor
+class LaundryViewModel: ObservableObject {
+    
+    @Published var laundryHallIds: ResultWithLoading<[LaundryHallId]> = .loading
+    @Published var hallUsages: [Int: ResultWithLoading<LaundryHallUsageResponse>] = [:] // key: hall_id
+    @Published var selectedHalls: Set<Int> = []
+    
+    @AppStorage("selectedLaundryHallIds") private var selectedHallsData: Data = Data()
+    
+    let maxSelection: Int = 3
+    
+    init() {
+        loadSelectedHalls()
+    }
+    
+    func loadLaundryHalls() async {
+        laundryHallIds = .loading
+        do {
+            let halls = try await LaundryAPIService.getLaundryHallIdData()
+            laundryHallIds = .success(halls)
+        } catch {
+            laundryHallIds = .failure(error)
+        }
+    }
+    
+    func loadLaundryHallUsage(for hallId: Int) async {
+        hallUsages[hallId] = .loading
+        do {
+            let usage = try await LaundryAPIService.getLaundryHallUsage(for: hallId)
+            hallUsages[hallId] = .success(usage)
+        } catch {
+            hallUsages[hallId] = .failure(error)
+        }
+    }
+    
+    func toggleSelection(_ hallId: Int) {
+        if selectedHalls.contains(hallId) {
+            selectedHalls.remove(hallId)
+        } else if canSelectMore(selectedHalls) {
+            selectedHalls.insert(hallId)
+        }
+        saveSelectedHalls()
+    }
+    
+    func setSelectedHalls(_ halls: Set<Int>) {
+        selectedHalls = halls
+        saveSelectedHalls()
+    }
+    
+    func removeSelectedHall(_ hallId: Int) {
+        selectedHalls.remove(hallId)
+        saveSelectedHalls()
+    }
+    
+    func isSelected(_ hallId: Int) -> Bool {
+        selectedHalls.contains(hallId)
+    }
+    
+    func canSelectMore(_ tempSelection: Set<Int>) -> Bool {
+        tempSelection.count < maxSelection
+    }
+    
+    func currentSelectedHalls() -> Set<Int> {
+        selectedHalls
+    }
+    
+    private func saveSelectedHalls() {
```

Comment:
If we replace this with a `didSet`-implementation, will this still work? Should save us from having to call `saveSelectedHalls()` repeatedly